core/src/main/java/org/apache/druid/math/expr/ApplyFunction.java

```diff
@@ -114,6 +150,14 @@ default boolean hasArrayOutput(LambdaExpr lambdaExpr)
   @Nullable
   ExpressionType getOutputType(Expr.InputBindingInspector inspector, LambdaExpr expr, List<Expr> args);
 
+  class ValidationException extends IllegalArgumentException
```

Comment:
Why does this class inherit `IllegalArgumentException` instead of `IAE`? I'm a little bit confused.